packages/bpk-component-datatable/src/BpkDataTable.js

```diff
@@ -0,0 +1,162 @@
+  const sorted = _sortBy(list, sortBy);
+  if (sortDirection === 'DESC') {
+    sorted.reverse();
+  }
+  return sorted;
+};
+
+class BpkDataTable extends Component {
+  constructor(props) {
+    super(props);
+
+    const sortBy = 'name';
+    const sortDirection = 'ASC';
+    const sortedList = sortList({ sortBy, sortDirection, list: props.rows });
+    const columns = props.children.slice();
+
+    if (props.dir === 'rtl') {
+      columns.reverse();
+    }
+
+    this.state = {
+      sortedList,
+      sortBy,
+      sortDirection,
+      columns,
+      rows: props.rows,
+      rowSelected: undefined,
+    };
+
+    this.rowClicked = this.rowClicked.bind(this);
+    this.rowStyle = this.rowStyle.bind(this);
+    this.sort = this.sort.bind(this);
+  }
+
+  componentWillReceiveProps({ rows }) {
+    if (rows !== this.state.rows) {
+      const { sortBy, sortDirection } = this.state;
+      const sortedList = sortList({ sortBy, sortDirection, list: rows });
+      this.setState({ rows, sortedList, rowSelected: undefined });
+    }
+  }
+
+  rowClicked({ index }) {
+    if (this.state.rowSelected === index) {
+      this.setState({ rowSelected: undefined });
+    } else {
+      this.setState({ rowSelected: index });
+    }
+    if (this.props.onRowClick !== undefined) {
+      this.props.onRowClick(this.state.rows[index]);
+    }
+  }
+
+  rowStyle({ index }) {
+    const classNames = [getClassName('bpk-data-table__row')];
+    if (this.state.rowSelected === index) {
+      classNames.push(getClassName('bpk-data-table__row--selected'));
+    }
+    if (index === -1) {
+      classNames.push(getClassName('bpk-data-table__headerRow'));
+    }
+    return classNames;
+  }
+
+  sort({ sortBy, sortDirection }) {
+    const sortedList = sortList({ sortBy, sortDirection, list: this.state.rows });
+
+    this.setState({ sortBy, sortDirection, sortedList });
+  }
+
+  renderTable(width) {
+    const {
+      sortedList,
+      sortDirection,
+      sortBy,
+      columns,
+    } = this.state;
+    const { height, dir } = this.props;
+
+    return (
+      <Table
+        className={getClassName('bpk-data-table')}
+        width={width}
+        height={height}
+        headerHeight={60}
+        rowHeight={60}
+        rowCount={sortedList.length}
+        rowGetter={({ index }) => sortedList[index]}
+        headerClassName={getClassName('bpk-data-table__headerColumn')}
+        rowClassName={this.rowStyle}
+        onRowClick={this.rowClicked}
+        sort={this.sort}
+        sortBy={sortBy}
+        sortDirection={sortDirection}
+        gridStyle={{ direction: dir }}
+      >
+        {
+        columns.map((child, index) => (
+          BpkColumn({ ...child.props, key: index })
+        ))
+        }
+      </Table>
+    );
+  }
+
+  render() {
+    if (this.props.width !== null) {
+      return this.renderTable(this.props.width);
+    }
+
+    return (
+      <AutoSizer disableHeight>{({ width }) => this.renderTable(width)}</AutoSizer>
```

Comment:
We can bind `this.renderTable` to `this` and avoid creating this redundant anonymous function

Reply:
you're missing the destructured `{ width }` (function signatures don't match) :/